File: java/1436-destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Set<String> set = new HashSet<>();
        for(List<String> path: paths){
            String city = path.get(0);
            set.add(city);
        }

        for(List<String> path: paths){
            String city = path.get(1);
            if(!set.contains(city))
                return city;
        }
        return "";
    }
}

File: kotlin/1436-destination-city.kt
class Solution {
    fun destCity(paths: List<List<String>>): String {
        var fromCity: HashSet<String> = hashSetOf()
        for (path in paths)
            fromCity.add(path[0])

        var res = paths[0][1]
        for (path in paths) {
            if (path[1] !in fromCity) {
                res = path[1]
                break
            }
        }

        return res
    }
}
